src/main/java/net/fabricmc/loader/config/Identifiable.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.config;
+
+import net.fabricmc.loader.api.config.exceptions.ConfigIdentifierException;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+public class Identifiable {
+	private final String id;
+
+	public Identifiable(@NotNull String namespace, @NotNull String name) {
```

Comment:
Heck no, this is not fabric api or minecraft

Reply:
No, it's not, but having a unique way to identify resources is useful in plenty of other contexts than Minecraft.

Reply:
We shouldn't assume everything has a name or a namespace. Say what prevents us to use `.` separated names to indicate unique locations, such as what java module names use?